thirdeye/thirdeye-pinot/src/main/resources/assets/javascript/DataService.js

```diff
@@ -131,34 +131,39 @@ DataService.prototype = {
       var response = this.postData(url, data);
     },
 
-    fetchGranularityForMetric: function (metricId) {
-      var url = "/data/agg/granularity/metric/"+metricId;
-      return this.getDataSynchronous(url);
+    fetchGranularityForMetric(metricId) {
+      const url = `/data/agg/granularity/metric/${metricId}`;
+      return this.getDataAsynchronous(url);
     },
 
-    fetchDimensionsForMetric : function(metricId) {
-      var url = "/data/autocomplete/dimensions/metric/"+metricId;
-      return this.getDataSynchronous(url);
+    fetchDimensionsForMetric(metricId) {
+      const url = `/data/autocomplete/dimensions/metric/${metricId}`;
+      return this.getDataAsynchronous(url);
     },
 
-    fetchFiltersForMetric : function(metricId) {
-      var url = "/data/autocomplete/filters/metric/" + metricId;
-      return this.getDataSynchronous(url);
+    fetchFiltersForMetric(metricId) {
+      const url = `/data/autocomplete/filters/metric/${metricId}`;
+      return this.getDataAsynchronous(url);
+    },
+
+    fetchMaxTimeForMetric(metricId) {
+      const url = `/data/maxDataTime/metricId/${metricId}`;
+      return this.getDataAsynchronous(url);
```

Comment:
Can you generalize these into a function with more variable values in the string? They are all so similar it almost seems redundant